packages/api/server.js

```diff
@@ -56,6 +71,65 @@ app.get('/api/repos', async (req, res) => {
     .end();
 });
 
+app.get('/githubauth', (req, res) => {
+  const url = 'http://github.com/login/oauth/authorize?client_id=' + process.env.CLIENT_ID + '&state=MUSTCHANGE&allow_signup=false';
+  res.status(302).redirect(url);
+});
+
+app.get('/api/callback', (req, res) => {
+  if (req.param('state') !== 'MUSTCHANGE') {
+    res.status(401).send('Error: invalid state!');
+    return;
+  }
+
+  request.post(
+    {
+      headers: { 'content-type': 'application/json', Accept: 'application/json' },
+      url: 'https://github.com/login/oauth/access_token',
+      body: JSON.stringify({
+        code: req.param('code'),
+        client_id: process.env.CLIENT_ID,
+        client_secret: process.env.CLIENT_SECRET,
+        state: 'MUSTCHANGE' // TODO: make this something better
+      })
+    },
+
+    async function (error, response, body) {
+      if (error) {
+        res.status(401).redirect('https://flaky-dev-staging.web.app');
+        return;
+      }
+      request.get(
+        {
+          headers: { 'content-type': 'application/json', 'User-Agent': 'flaky.dev', Authorization: 'token ' + JSON.parse(body).access_token },
+          url: 'https://api.github.com/user'
+        },
+
+        await async function (error, response, body) {
+          if (error || JSON.parse(body).message === 'Bad credentials') {
+            res.status(401).redirect('https://flaky-dev-staging.web.app');
+            return;
+          }
+          const repository = new Repository();
+          const permitted = await repository.mayAccess('github', JSON.parse(body).login);
+          await repository.storeSessionPermission(req.sessionID, permitted);
+          res.status(200).redirect('https://flaky-dev-staging.web.app');
+        }
+      );
+    }
+  );
+});
+
+app.get('/sessionauthcheck', async (req, res) => {
+  const repository = new Repository();
+  const result = await repository.sessionPermissions(req.sessionID);
```

Comment:
I would be tempted to call this `/session`, and then you could return the `username` currently logged in, which could be used to display this contextual information to the frontend.